### PR TITLE
Rename event id to event hash

### LINKED ISSUE #1142: Rename eventId to eventHash in Execution
The `eventID` parameter should be renamed by `eventHash` in the Execution API and  also in Execution struct. The type should also be changed to `hash.Hash`.

### PR DESCRIPTION
**Pull request by krhubert**
close #1142

### CODE REVIEW

File: hash/hash_test.go [hunk 1]
```diff
@@ -25,6 +27,27 @@ func TestInt(t *testing.T) {
 	assert.Len(t, Int(1), size)
 }
 
+func TestRnadom(t *testing.T) {
```
Comment (NicolasMahe) [inline hash/hash_test.go]:
```suggestion
func TestRandom(t *testing.T) {
```

File: hash/hash.go [hunk 2]
```diff
@@ -58,6 +59,21 @@ func Int(h int) Hash {
 	return hash
 }
 
+// Random returns a new random hash.
+func Random() (Hash, error) {
```
Comment (NicolasMahe) [inline hash/hash.go]: Should use this function is way more test ;)

File: execution/execution.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func (s Status) String() (r string) {
 type Execution struct {
 	Hash         hash.Hash              `hash:"-"`
 	ParentHash   hash.Hash              `hash:"name:parentHash"`
-	EventID      string                 `hash:"name:eventID"`
+	EventHash    hash.Hash              `hash:"name:eventID"`
```
Comment (NicolasMahe) [inline execution/execution.go]:
```suggestion
	EventHash    hash.Hash              `hash:"name:eventHash"`
```

File: hash/hash.go (hunk 2 above)
Comment (krhubert) [reply hash/hash.go]: Int is fine :P

File: hash/hash_test.go [hunk 4]
```diff
@@ -25,6 +27,27 @@ func TestInt(t *testing.T) {
...
+			return false
+		}
+
+		if seen[hash.String()] {
+			return false
+		}
+		seen[hash.String()] = true
+		return true
+	}
+	require.NoError(t, quick.Check(f, nil))
```
Comment (antho1404) [inline hash/hash_test.go]: didn't know about this quick method 👍

**Pull request merged**